### ISSUE TITLE
Spring boot project : jpa query returning old data

### ISSUE THREAD

**Issue by ankurgarg1041** [spring-projects/spring-boot#6637]
Hi,
We have created a spring boot project using 1.3.5 version. Our application interacts with Mysql database.
We have created a set of jpa-repositories in which we are using findAll, findOne and other custom queries method.

We are facing a issue which is occurring randomly  .  Following are the  steps to reproduce it:
1. Fire a read query on db using spring-boot application.
2. Now Manually  change the data in Mysql  using mysql-console  for the records which are returned by above read query.
3. Now again fire the same read query using application.
After step 3 , we should have received  the modified results of step 2, but what we got was the data before modification.

4. Now if we again fire the read query using application, It gives us correct values.

This issue is happening randomly. We are not using any kind of cache in our application.

While debugging I found out that jpa-repository code is calling database and it fetches the latest result .  But when this call return back to our application service , it has the old data.

Please help us identify the possible cause of it.

**Comment by snicoll** [comment]
Thanks for getting in touch, but it feels like this is a question that would be better suited to [Stack Overflow](http://stackoverflow.com/). As mentioned in [the guidelines for contributing](https://github.com/spring-projects/spring-boot/blob/master/CONTRIBUTING.adoc#using-github-issues), we prefer to use GitHub issues only for bugs and enhancements. Feel free to update this issue with a link to the re-posted question (so that other people can find it) or add some more details if you feel this is a genuine bug.

**Comment by ankurgarg1041** [comment]
Hi
Let me rephrase my concern.
My concern is that is there a any chance of possible bug in spring boot around this ?

Has any one reported similar kind of bug till date?

First I thought that there might be an issue at my end but I  have been contacted by an another person in my organisation who is facing similar issue with their application.

Please let me know if still I should post it only on stackoverflow.

**Comment by snicoll** [comment]
> Please let me know if still I should post it only on stackoverflow.

Yes.

**Comment by pmandapuram** [comment]
I have the similar problem. 
If anyone already reported the same issue please post the link below for better communication to solve the issue.
Thanks.
